parser/src/main/scala/hmda/parser/fi/FIDataCsvParser.scala

```diff
@@ -13,8 +13,9 @@ class FIDataCsvParser extends FIDataParser[String] {
   def parseLines(lines: Iterable[String]): FIData = {
     val tsLine = lines.head
     val ts = TsCsvParser(tsLine)
-    val lars = lines.tail.map(l => LarCsvParser(l))
-    FIData(ts, lars)
+    // Only parse LARs that do not contain errors
+    val larsWithoutErrors = lines.tail.map(l => LarCsvParser(l)).filter(_.isRight).map(_.right.get)
+    FIData(ts, larsWithoutErrors)
```

Comment:
I like the idea of creating a separate issue for this FIData behavior, so we can implement this first. I suspect the issue of parsing+validating whole files will be large, so it's good to reduce its scope where we can.